### PR TITLE
Avoid performing tilde expansion

### PR DESCRIPTION
**Pull request by warwickmm**
The expansion being performed was often incorrect, as it's possible for the tilde character to appear as part of the path where it should not be expanded to the user's home directory.

Properly supporting tilde expansion is complicated and requires additional study.  Instead of supporting it only partially, we have decided to simplify things and just drop tilde expansion for now.  See the discussion in issue #2619, as well as issues #2325 and #2555.

### CODE REVIEW

File: Duplicati/Library/Utility/Utility.cs [hunk 1]
```diff
@@ -974,16 +974,16 @@ public static string LocateFileInSystemPath(string filename)
...
-        private static readonly string HOME_PATH = Environment.GetFolderPath(IsClientLinux ? Environment.SpecialFolder.Personal : Environment.SpecialFolder.UserProfile);
+        public static readonly string HOME_PATH = Environment.GetFolderPath(IsClientLinux ? Environment.SpecialFolder.Personal : Environment.SpecialFolder.UserProfile);
 
         /// <summary>
-        /// Expands environment variables, including the tilde character
+        /// Expands environment variables.
         /// </summary>
         /// <returns>The expanded string.</returns>
         /// <param name="str">The string to expand.</param>
         public static string ExpandEnvironmentVariables(string str)
```
Comment (warwickmm) [inline Duplicati/Library/Utility/Utility.cs]: This method is no longer necessary, as we can just call the built-in method.  I decided to leave this refactoring for a separate pull request so that we don't pollute the diff in this pull request.

**Comment by warwickmm** [comment]
@kenkendk, you [previously mentioned](https://github.com/duplicati/duplicati/issues/2619#issuecomment-333094921) that `~` characters may be saved in the configuration and that an upgrade script may be needed.  Is this still a concern?

File: Duplicati/Library/Utility/Utility.cs (hunk 1 above)
Comment (warwickmm) [reply Duplicati/Library/Utility/Utility.cs]: Actually, it seems that `ExpandEnvironmentVariables` will only expand Windows-style environment variables (e.g., it will expand `%HOME%` but not `$HOME`).  Do we need to better support Linux and MacOS here?

Comment (kenkendk) [reply Duplicati/Library/Utility/Utility.cs]: Yes, I have a comment about that in the code. Until now it has only used the Windows-style environment variable expansion (that is how .Net does) but it is simple to do the Linux style expansion as well.

Should we just drop the old Windows-style environment variable expansion on non-Windows? I am not sure.... I think it may break something for current users expecting the expansion, but new users will be confused if it does not work as all the other stuff on their system....

**Comment by kenkendk** [comment]
Yes, I think it is a potential problem. The source paths for the backup job is saved in the database. If we no longer expand them, we should run through all saved (source)paths and apply the expansion.

It could still be problematic for CLI users, but I expect them to be better at dealing with the issue.

**Comment by kenkendk** [comment]
For the build error, if you merge `master` into your branch, it should fix the build issue on Travis.

File: Duplicati/Library/Utility/Utility.cs (hunk 1 above)
Comment (warwickmm) [reply Duplicati/Library/Utility/Utility.cs]: I'm not sure what the best approach is.  Is the main purpose here to expand environment variables specified in command-line options and filters?  If so, one option is to only support (and document) the Windows-style syntax.  Unless there's a way to support both on non-Windows environments if we want to avoid breaking current functionality?

I'll create a separate issue regarding this matter so we can continue the discussion there.

**Comment by warwickmm** [comment]
> Yes, I think it is a potential problem. The source paths for the backup job is saved in the database. If we no longer expand them, we should run through all saved (source)paths and apply the expansion.

Unfortunately, I'm not familiar enough with the code to do address this.  I'd appreciate any help/hints on this. :confused: 

> For the build error, if you merge master into your branch, it should fix the build issue on Travis.

I merged in 43d9b51743, but 2 of the Travis jobs failed with:

```No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.```

I'm not sure how to diagnose this...did my changes break the build somehow?

File: Duplicati/Library/Utility/Utility.cs (hunk 1 above)
Comment (warwickmm) [reply Duplicati/Library/Utility/Utility.cs]: Created issue #2838.

**Comment by kenkendk** [comment]
Sorry for the delay.

The build error is a Travis issue, no worries.

I have not merged this yet because I need to find the time to look into the potential storage of paths with environment variables and figure out if/how this should be handled.

**Comment by warwickmm** [comment]
No worries.  Sorry for the incomplete pull request...I'm still slowly finding my way around the code.

**Comment by kenkendk** [comment]
Ok, found it. It actually uses `%HOME%` so no problems with that.
If we fix #2838 we need to take care to update the variables in the database as well.
